packages/daemon/src/async-hooks.js

```diff
@@ -0,0 +1,14 @@
+/**
+ * @returns {import('./types.js').AsyncHooks<any>}
+ */
+export const makeAsyncHooks = () => {
```

Comment:
There’s a Node.js module by the same name that we don’t want to confuse this with. This is similar to a `WaitGroup` in Go. I think `DeferredSet` might be good.

Reply:
I went with `deferredTasks`.